### PR TITLE
Add few patch types and rewrite fs::get_parent_dir

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

File: Utilities/bin_patch.h [hunk 1]
```diff
@@ -34,9 +34,12 @@ enum class patch_type
 	lef64,
 	be16,
 	be32,
+	bd32,
```
Comment (Megamouse) [inline Utilities/bin_patch.h:37]: what do these 3 new types stand for ?

File: Utilities/bin_patch.cpp [hunk 2]
```diff
@@ -427,6 +430,11 @@ bool patch_engine::add_patch_data(YAML::Node node, patch_info& info, u32 modifie
 
 	switch (p_data.type)
 	{
+	case patch_type::utf8:
+	{
+		p_data.value.long_value = p_data.original_value.size();
```
Comment (Megamouse) [inline Utilities/bin_patch.cpp:435]: I think this line can be removed, as long_value is not used anywhere for utf8

Comment (Nekotekina) [reply Utilities/bin_patch.cpp:435]: hmm, right, it's unused

File: Utilities/bin_patch.h (hunk 1 above)
Comment (Nekotekina) [reply Utilities/bin_patch.h:37]: 2 types are hints that they don't contain executable code, I tried to explain in commit message.

File: Utilities/File.cpp [hunk 3]
```diff
@@ -322,12 +322,53 @@ std::shared_ptr<fs::device_base> fs::set_virtual_device(const std::string& name,
 
 std::string fs::get_parent_dir(const std::string& path)
 {
-	// Get (basically) processed path
-	const auto real_path = fs::escape_path(path);
```
Comment (elad335) [inline Utilities/File.cpp:326]: maybe fix escape_path to work with network locations.

File: Utilities/bin_patch.h (hunk 1 above)
Comment (Megamouse) [reply Utilities/bin_patch.h:37]: then maybe add a comment in code for both

File: Utilities/File.cpp (hunk 3 above)
Comment (Nekotekina) [reply Utilities/File.cpp:326]: Parent dir always worked scanning from the end, and it was OK for mimimal purposes. The problem with renaming arises not so often and there are different approaches to it and other problems (such as case sensitivity). I have one fundamental in mind which I mentioned as "journaled VFS".

**Pull request merged**